### PR TITLE
fix: follow GraphQL over HTTP in error handling

### PR DESCRIPTION
**Pull request by StarpTech**
## Motivation and Context

- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [ ] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

### CODE REVIEW

File: router/core/graphql_prehandler.go [hunk 1]
```diff
@@ -132,6 +162,9 @@ func (h *PreHandler) Handler(next http.Handler) http.Handler {
 
 		statusCode = ww.Status()
 		writtenBytes = ww.BytesWritten()
+
+		// Evaluate the request after the request has been handled by the engine
+		hasRequestError = requestContext.hasError
```
Comment (StarpTech) [inline router/core/graphql_prehandler.go:167]: @fiam in order to do this I had to move the creation of the context before the graphql handler.

File: router/core/websocket.go [hunk 2]
```diff
@@ -463,13 +470,42 @@ func (h *WebSocketConnectionHandler) executeSubscription(ctx context.Context, ms
 	// This gets removed by WebSocketConnectionHandler.Complete()
 	h.subscriptions.Insert(msg.ID, cancel)
 
-	ctxWithOperation := withOperationContext(cancellableCtx, operation, h.clientInfo)
+	variablesCopy := make([]byte, len(operation.Variables))
```
Comment (StarpTech) [inline router/core/websocket.go:473]: This is duplicated in order to freely modify the request context in both places.

**Review by fiam — APPROVED** [review]
LGTM

File: router/core/context.go [hunk 3]
```diff
@@ -424,3 +397,29 @@ func subgraphsFromContext(ctx context.Context) []Subgraph {
...
+		content:    operation.NormalizedRepresentation,
+		hash:       operation.ID,
+		variables:  variablesCopy,
+		clientInfo: clientInfo,
+	}
+
+	subgraphs := subgraphsFromContext(r.Context())
+	requestContext := &requestContext{
+		logger:         requestLogger,
+		keys:           map[string]any{},
```
Comment (jensneuse) [inline router/core/context.go:417]: What are keys for?

**Review by jensneuse — APPROVED** [review]

File: router/core/context.go (hunk 3 above)
Comment (StarpTech) [reply router/core/context.go:417]: A bucket to share data across modules e.g. `ctx.MustGet('foo')`

**Pull request merged**